core/types/calls.cc

```diff
@@ -495,14 +495,30 @@ DispatchResult dispatchCallSymbol(Context ctx, DispatchArgs args,
                     vector<ErrorLine> lines;
                     lines.reserve(alternatives.size());
                     for (auto alternative : alternatives) {
-                        auto possible_symbol = alternative.symbol.data(ctx);
-                        if (!possible_symbol->isClass() && !possible_symbol->isMethod()) {
+                        auto possibleSymbol = alternative.symbol.data(ctx);
+                        if (!possibleSymbol->isClass() && !possibleSymbol->isMethod()) {
                             continue;
                         }
-                        auto suggestedName = possible_symbol->isClass() ? alternative.symbol.show(ctx) + ".new"
-                                                                        : alternative.symbol.show(ctx);
+
+                        auto suggestedName = possibleSymbol->isClass() ? alternative.symbol.show(ctx) + ".new"
+                                                                       : alternative.symbol.show(ctx);
                         lines.emplace_back(
                             ErrorLine::from(alternative.symbol.data(ctx)->loc(), "Did you mean: `{}`?", suggestedName));
+
+                        if (possibleSymbol->isClass()) {
+                            const auto replacement = possibleSymbol->name.show(ctx);
+                            const auto loc = args.locs.call;
+                            const auto methodLoc = Loc{loc.file(), loc.beginPos(),
+                                                       (u4)(loc.beginPos() + args.name.toString(ctx).length())};
+                            e.replaceWith(fmt::format("Replace with `{}.new`", replacement), methodLoc, "{}.new",
+                                          replacement);
+                        } else {
+                            const auto replacement = possibleSymbol->name.toString(ctx);
+                            const auto loc = args.locs.receiver;
+                            const auto methodLoc = Loc{loc.file(), loc.endPos() + 1,
+                                                       (u4)(loc.endPos() + 1 + args.name.toString(ctx).length())};
```

Comment:
there is no guarantee that the `name` of send has anything to do with the textual send.
This will create invalid(and out of bounds) accesses for desugared code.

Please, don't do math on `loc`s

Reply:
Yeah, this did not feel great but I couldn't figure out how to get the method call's location otherwise (and partially of cargo culted this mathy approach from https://github.com/sorbet/sorbet/blob/master/core/types/calls.cc#L945). What's the alternative?

Reply:
the difference is that one is that the other one makes a smaller loc, rather than a bigger one. Going behind `loc.endPos` is the risky part